wordpress-mobile/WordPress-Android
# Added filtering by supported MIME types in device media choose intents

Fixes #12858 

Related FluxC PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1677
_**Note**: will update the FluxC commit version to a proper version tag once the FluxC PR gets approved and merged_

This PR adds support to not allow introducing unsupported MIME type items from the device media choosers. This same filtering is being used already in the MediaBrowserActivity by using `WPMediaUtils.launchFileLibrary()` in [this line](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java#L990). It internally uses this:
```
intent.putExtra(Intent.EXTRA_MIME_TYPES, new MimeTypes().getAllTypes());
```

So basically this PR uses the same approach to filter out by only the supported mime types for video and images (and both when the case merits) only.

#### To test
Before all tests, download an unsupported media type to your test phone so we can test it gets filtered out in each case.
- For example, download a webp or another WP-unsupported media file on Chrome on your device / emulator. I used the examples given here https://developers.google.com/speed/webp/gallery1
- For videos, I downloaded a webm file from https://file-examples.com/index.php/sample-video-files/sample-webm-files-download/

Aztec:
1. make sure to set "Use block editor" to **OFF** in your site settings
2. tap on the FAB and then new Post
3. tap on the cross on top of the keyboard to open the media picker
4. now tap on the device chooser icon
5. tap on Choose video from device / Choose image from device
6. observe the unsupported mime type files appear greyed out
![aztecpicker](https://user-images.githubusercontent.com/6597771/91863928-068d2280-ec46-11ea-9fe9-9b28dc0bba8f.gif)

Gutenberg:
1. make sure to set "Use block editor" to **ON** in your site settings
2. tap on the FAB and then new Post
3. add an image block, then tap on ADD IMAGE / add a video block, then tap on ADD VIDEO
4. on the bottom sheet, tap on "Choose from device"
5. in the media picker, tap on the bottom button to trigger the device's chooser
<img width="308" alt="Screen Shot 2020-09-01 at 11 28 47" src="https://user-images.githubusercontent.com/6597771/91864212-553abc80-ec46-11ea-9b19-6431063072bb.png">
6. observe the unsupported mime type files appear greyed out
![gutenbergpicker](https://user-images.githubusercontent.com/6597771/91864396-89ae7880-ec46-11ea-9955-68bc03888422.gif)

Stories:
1. tap on the FAB and then new Story
2. when the media picker shows up, tap on the bottom left button
4. on the bottom sheet, tap on "Choose photo from device" / "Choose video from device"
<img width="301" alt="Screen Shot 2020-09-01 at 11 31 45" src="https://user-images.githubusercontent.com/6597771/91864602-c5494280-ec46-11ea-959f-0b2aa10f44b1.png">
5. observe the unsupported mime type files appear greyed out
![storiespicker](https://user-images.githubusercontent.com/6597771/91864684-e316a780-ec46-11ea-993f-3b24d6e909aa.gif)


PR submission checklist:
